### PR TITLE
Add epsilon argument to Adagrad optimizer

### PR DESCRIPTION
**Pull request by li-roy**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24980 Add epsilon argument to Adagrad optimizer**

We'll need this internally, so just updating the open source version. the other optimizers have this argument anyways.

Differential Revision: [D16945279](https://our.internmc.facebook.com/intern/diff/D16945279)

### CODE REVIEW

File: torch/optim/adagrad.py [hunk 1]
```diff
@@ -12,24 +12,28 @@ class Adagrad(Optimizer):
...
+            numerical stability (default: 1e-10)
         lr_decay (float, optional): learning rate decay (default: 0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
 
     .. _Adaptive Subgradient Methods for Online Learning and Stochastic
         Optimization: http://jmlr.org/papers/v12/duchi11a.html
     """
 
-    def __init__(self, params, lr=1e-2, lr_decay=0, weight_decay=0, initial_accumulator_value=0):
+    def __init__(self, params, lr=1e-2, lr_decay=0, eps=1e-10, weight_decay=0, initial_accumulator_value=0):
```
Comment (bddppq) [inline torch/optim/adagrad.py]: better to put new arg to the end (for backward compatibility)

**Review by bddppq — COMMENTED** [review]
LGTM

**Review by bddppq — APPROVED** [review]
LGTM

**Review by wanchaol — APPROVED** [review]

File: torch/optim/adagrad.py (hunk 1 above)
Comment (li-roy) [reply torch/optim/adagrad.py]: I ordered them this way initially to match the other optimizers, but yeah putting it at the end is better for bc.

**Comment by ezyang** [comment]
cc'ing Vincent who has been looking at optimizers a lot

**Comment by ezyang** [comment]
@vincentqb You should consider adding yourself to module: optimizer on https://github.com/pytorch/pytorch/issues/24422

**Pull request closed without merging**